parcels/_core/utils/string.py

```diff
@@ -0,0 +1,9 @@
+from keyword import iskeyword, kwlist
+
+
+def _assert_valid_python_varname(name):
+    if name.isidentifier() and not iskeyword(name):
+        return
+    raise ValueError(
+        f"Received invalid Python variable name {name!r}. Avoid using the following names: {', '.join(kwlist)}"
+    )
```

Comment:
I think including the kwlist error message in the output everytime is perhaps excessive - try the following?

```suggestion
def _assert_valid_python_varname(name):
    if name.isidentifier() and not iskeyword(name):
        return
    msg = f"Received invalid Python variable name {name!r}. "
    if name in kwlist:
        msg += f"HINT: Avoid using the following names: {', '.join(kwlist)}"
    raise ValueError(msg)
```